nexus/types/src/deployment/planning_input.rs

```diff
@@ -466,10 +475,32 @@ impl PlanningInputBuilder {
         }
     }
 
+    /// Like `add_omicron_zone_external_ip`, but can accept an [`IpNetwork`],
+    /// validating that the IP is a single address.
+    pub fn add_omicron_zone_external_ip_network(
+        &mut self,
+        zone_id: OmicronZoneUuid,
+        ip_id: ExternalIpUuid,
+        ip: IpNetwork,
+    ) -> Result<(), PlanningInputBuildError> {
+        let size = match ip.size() {
+            NetworkSize::V4(n) => n as u128,
```

Comment:
Could we use 

```suggestion
            NetworkSize::V4(n) => u128::from(n),
```

instead? `as` has always made me skittish.

Reply:
good point -- in this case it's okay because it's lossless. Wonder if we should activate some of the lints in https://rust-lang.github.io/rust-clippy/master/index.html#/cast.

Reply:
Added a clippy warning in https://github.com/oxidecomputer/omicron/pull/5544.

Reply:
And fixed in this commit separately as well.